src/term/mod.rs

```diff
@@ -1271,6 +1308,38 @@ mod tests {
     use index::{Point, Line, Column};
     use term::{Cell};
     use ansi::{Handler, CharsetIndex, StandardCharset};
+    use selection::Selection;
+    use std::mem;
+
+    /// pub fn semantic_selection(&self, selection: &mut Selection, point: Point) {
+    #[test]
+    fn semantic_selection_works() {
+        let size = SizeInfo {
+            width: 21.0,
+            height: 51.0,
+            cell_width: 3.0,
+            cell_height: 3.0,
+        };
+        let mut term = Term::new(&Default::default(), size);
+        let mut grid: Grid<Cell> = Grid::new(Line(5), Column(1), &Cell::default());
+        for i in 0..5 {
+            grid[Line(i)][Column(0)].c = 'a';
+        }
+        grid[Line(0)][Column(0)].c = '"';
+        grid[Line(4)][Column(0)].c = '"';
+
+        let mut escape_chars = String::from("\"");
+
+        mem::swap(&mut term.grid, &mut grid);
+        mem::swap(&mut term.semantic_escape_chars, &mut escape_chars);
+
+        let mut selection = Selection::new();
+        term.semantic_selection(&mut selection, Point { line: Line(3), col: Column(0) });
+        match selection.span() {
+            Some(span) => assert_eq!(term.string_from_selection(&span), "a\na\na"),
```

Comment:
Hmm, should double click span multiple lines? I was under the impression that double click would only ever match within a line and that triple click should expand across multiple lines *if* the terminal had wrapped a long line.

Basically, I guess we didn't talk about how semantic selection could behave. We should embrace the principal of least surprise here and attempt to match behavior from other terminal emulators.